src/reducers/app.js

```diff
@@ -227,10 +227,15 @@ function updateDimensions(appState, action) {
 }
 
 function toggleLanguage(appState, action) {
-  const otherLanguage = appState.language === "es-MX" ? "en-US" : "es-MX";
-  return Object.assign({}, appState, {
-    language: action.language || otherLanguage,
-  });
+  return {
+    ...appState,
+    language: action.language || selectNextLanguage(appState),
+  };
+  function selectNextLanguage({ language, languages }) {
+    const currentIndex = appState.languages.indexOf(language);
+    const nextIndex = (currentIndex + 1) % languages.length;
+    return languages[nextIndex];
```

Comment:
This toggle strategy is interesting but I'd say it does not scale as intended if we add more languages, I'd prefer if we went with a component visually similar to the ones for Filters (and fullscreen) which would open a menu with the language options, do you think you could refactor it in that way?

Reply:
Yeah. I'll be happy to take it further that way.

I'll appreciate some design suggestions. I see two key pieces that are needed here, but only vague idea of how exactly they should look.

1. Language-independent indicator in the toolbar, indicating the place to click to change the language. 
2. The language switch in an opened menu (rather large menu, almost full screen on the mobile)

The indicator could be an icon and/or text label with letters that are unique to each of the primary alphabets (like `LЪҐ`, `SЭЄ`, `WЫЇ`). The language switch could be showing separate buttons for each language, saying "In English", "На русском", ?"На українській"?

These components could be placed either as
- a separate toolbar entry with it's own "menu", or
- combined with the toolbar title (top left corner) and the big "about" screen. Language-change indicator right under the "civilian harm in Ukraine title" and the language switch buttons on top of "About and Methodology" header.

Reply:
I like the idea of having a toolbar button (1st from the top), liek so:

![image](https://user-images.githubusercontent.com/19508417/162076752-926127cb-3754-4317-a9a7-a5ec56f5122e.png)
(or other relevant text)

but I'm also happy with a slightly different look for example right under the "Civilian Harm in Ukraine".

---
As for your point 2. language switch, I agree with a menu/panel with one entry/button with text/chars in the specific language - could also go for some minimalist icons for the flags [like duolingo's](https://preview.redd.it/hq9q08mrkct31.png?width=960&crop=smart&auto=webp&s=5da240927ba3205117893a23a8d7e2f4e7b77b0f) but that's totally optional (it's good that you're thinking ahead for mobile but that part is harder to do atm since we don't have the toolbar 😓).